File: Ser2023/Company/Company.Web/Modules/Company/ContactType/ContactTypeColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.ContactType")]
[BasedOnRow(typeof(ContactTypeRow), CheckNames = true)]
public class ContactTypeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long ContactTypeId { get; set; }
    [EditLink]
    public string Name { get; set; }
    //public Guid Rowguid { get; set; }
    //public short IsActive { get; set; }
    //public int TenantId { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/ContactTypeColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { ContactTypeRow } from './ContactTypeRow';

export interface ContactTypeColumns {
    ContactTypeId: Column<ContactTypeRow>;
    Name: Column<ContactTypeRow>;
    IsActive: Column<ContactTypeRow>;
    TenantId: Column<ContactTypeRow>;
}

export class ContactTypeColumns extends ColumnsBase<ContactTypeRow> {
    static readonly columnsKey = 'Company.ContactType';
    static readonly Fields = fieldsProxy<ContactTypeColumns>();
}